src/Model/Release/ComposerConstraint.php

```diff
@@ -16,6 +16,13 @@
  */
 class ComposerConstraint
 {
+    /**
+     * Used to hack in dual support for graphql on CMS 4
+     * Update this whenever a new version of graphql 3 is released
+     */
+    private const GRAPHQL_3_VERSION_DEV = '3.8.x-dev';
+    private const GRAPHQL_3_VERSION_STABLE = '~3.8.1@stable';
```

Comment:
I feel like it would be too easy to forget to update this if we release a new patch.
How important is that `.1` there as opposed to `.0`? If we're going to specify the exact patch version I think we should get that from the github api instead of hardcoding it.

Reply:
In practice it makes no difference.

Using the github api from ComposerConstraint would be nicer, but it requires a lot of code change to do this.
There's actually 2x github api clients
- GithubApi (old, seems unused)
- PublishRelease::getGithubClient() (the one that is actually used e.g. creating github releases)

Unfortunately PublishRelease::getGithubClient() cannot easily be separated (I tried) due to it needing to instantiate a StepCommandRunner, which itself needs a Step.  It's basically just needs a solid refactor in order to use it.

The chance of graphql 3.9 being released at this stage seems fairly remote.  And if it is, it'll be in the context of a security release, and I'd expect we'd be pretty conscious of checking the what's released in the CMS 4 version of recipe-cms at that stage.

Reply:
> In practice it makes no little difference if it's .1 or .0. It would only affect a --prefer-lowest CI build.

Let's set it to `.0` then to make the intention clearer (i.e. we don't care that patch version is, what we care about is that it's stable)

Reply:
`.1` is better, it could make a difference on `--prefer-lowest` builds for CMS 4

When I say "in practice" - no-one other than our CI use --prefer-lowest.  It's still nice to have for our own purposes.

Reply:
Then I revert back to my original position. If we care about the patch version we should just the API to get the correct patch version, since we're not likely to remember to update that as we do patch releases.

Reply:
As mentioned above I think there's too much refactoring required at this stage to get this working.  We can open a new card if you think it's worthwhile to look at later

Reply:
This hasn't changed since last time it was assigned to me.... I'll just let it stay as is, it's not a big deal.